File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum  =0;
        int maxSum =nums[0];
        if(nums.length == 1) return nums[0];

        for(int num : nums)
        {
            if(sum < 0) sum = 0;
            sum += num;
            maxSum = Math.max(sum,maxSum);
        }

        return maxSum;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        maxSum = nums[0]
        sum = 0
        for i in range(len(nums)):
            
            # for j in range(i,len(nums)):
            #     sum += nums[j]
            #     maxSum = max(maxSum,sum)

            if sum < 0: sum = 0
            sum += nums[i]

            maxSum = max(maxSum,sum)

        return maxSum
